### PR TITLE
Wait for mysql to be really available

### PR DESCRIPTION
**Pull request by meatballhat**
up to the default travis_wait timeout, plus reworking how service overrides work
a smidge.

### CODE REVIEW

File: lib/travis/build/appliances/services.rb [hunk 1]
```diff
@@ -35,6 +31,25 @@ def apply?
...
+        end
+
+        def apply_mysql
+          sh.raw <<~BASH
+            travis_mysql_ping() {
+              mysql &>/dev/null <<<'SHOW VARIABLES like "%version%"'
+            }
+          BASH
+          sh.cmd 'sudo service mysql start', assert: false, echo: true, timing: true
+          sh.cmd 'travis_wait travis_mysql_ping', assert: false, echo: false, timing: false
```
Comment (BanzaiMan) [inline lib/travis/build/appliances/services.rb]: I am not sure if `travis_wait` is a good choice here. The smallest granularity that it gives is 1 minute, which might be too coerce. In addition, I am thinking about tweaking `travis_wait` a bit; have an idea, but have not worked on it.

A more straightforward thing to do is to check if the TCP port is open with some timeout.

```sh
until ruby -rsocket -e "TCPSocket.new('localhost',3306)" || test $count -lt 5; do sleep 1; let count=$count+1; done
```

Comment (BanzaiMan) [inline lib/travis/build/appliances/services.rb]: See #975 for the `travis_wait` tweak.

Comment (meatballhat) [inline lib/travis/build/appliances/services.rb]: @BanzaiMan How should I modify this PR, if at all? 😁

Comment (BanzaiMan) [inline lib/travis/build/appliances/services.rb]: @meatballhat I suggest dropping the `travis_mysql_ping` function definition, and replacing this command `travis_wait travis_mysql_ping` with what I've suggested. (There should be a more succinct way to test TCP port 3306, though, than using `ruby`.)

Comment (emdantrim) [inline lib/travis/build/appliances/services.rb]: any action on this? this PR is very stale :crying_cat_face:

**Comment by meatballhat** [comment]
Superseded by #1121

**Pull request closed without merging**